src/theory/builtin/type_enumerator.h

```diff
@@ -75,6 +76,49 @@ class UninterpretedSortEnumerator : public TypeEnumeratorBase<UninterpretedSortE
 
 };/* class UninterpretedSortEnumerator */
 
+
+/** FunctionEnumerator 
+* This enumerates function values, based on the enumerator for the
+* array type corresponding to the given function type.
+*/
+class FunctionEnumerator : public TypeEnumeratorBase<FunctionEnumerator> {
+public:
+  FunctionEnumerator(TypeNode type, TypeEnumeratorProperties * tep = NULL) :
```

Comment:
What is the memory lifetime of tep expected to be?

Reply:
tep is not owned by the TypeEnumerator.  I've added some comments in theory/type_enumerator.h to make this clear.